src/test/java/PostgresDriverWrapperTest.java

```diff
@@ -0,0 +1,141 @@
+      private int iterations = 100;
+
+      @Override
+      public Object answer(InvocationOnMock invocationOnMock) throws Throwable {
+        return iterations-- > 0;
+      }
+    });
+
+    String userName = "postgres";
+    String password = "postgres";
+    String urlPort = "localhost:5432";
+    this.postgresDriverWrapper = new PostgresDriverWrapper(urlPort, userName, password);
+  }
+
+  @AfterEach
+  void tearDown() throws Exception {
+    postgresDriverWrapper.close();
+  }
+
+  @Test
+  void testConnect() throws SQLException {
+    try (MockedStatic<DriverManager> driverManagerMockedStatic = Mockito.mockStatic(DriverManager.class)) {
+      driverManagerMockedStatic.when(() -> DriverManager.getConnection(Mockito.any(), Mockito.any()))
+          .thenReturn(mockConnection);
+      Mockito.when(mockConnection.isClosed()).thenReturn(false);
+      postgresDriverWrapper.connect();
+      Assertions.assertTrue(postgresDriverWrapper.isConnected());
+    }
+  }
+
+  @Test
+  void testExecuteQueryAll() throws SQLException {
+    try (MockedStatic<DriverManager> driverManagerMockedStatic = Mockito.mockStatic(DriverManager.class)) {
+      // Mocks setup
+      driverManagerMockedStatic.when(() -> DriverManager.getConnection(Mockito.any(), Mockito.any()))
+          .thenReturn(mockConnection);
+      String sqlQuery = "select * from personaldata";
+      Mockito.when(mockConnection.prepareStatement(sqlQuery)).thenReturn(mockPreparedStatement);
+      Mockito.when(mockResultSet.getInt(Mockito.any())).thenReturn(1);
+      Mockito.when(mockResultSet.getDouble(Mockito.any())).thenReturn(1.0);
+      Mockito.when(mockResultSet.getString(Mockito.any())).thenReturn("test");
+      Mockito.when(mockResultSetMetadata.getColumnCount()).thenReturn(4);
+      Mockito.when(mockResultSetMetadata.getColumnName(1)).thenReturn("id");
+      Mockito.when(mockResultSetMetadata.getColumnName(2)).thenReturn("name");
+      Mockito.when(mockResultSetMetadata.getColumnName(3)).thenReturn("numberrange");
+      Mockito.when(mockResultSetMetadata.getColumnName(4)).thenReturn("currency");
+      Mockito.when(mockResultSetMetadata.getColumnTypeName(1)).thenReturn("serial");
+      Mockito.when(mockResultSetMetadata.getColumnTypeName(2)).thenReturn("varchar");
+      Mockito.when(mockResultSetMetadata.getColumnTypeName(3)).thenReturn("float8");
+      Mockito.when(mockResultSetMetadata.getColumnTypeName(4)).thenReturn("varchar");
+
+      postgresDriverWrapper.connect();
+      try (VectorSchemaRoot vectorSchemaRoot = postgresDriverWrapper.executeQuery(sqlQuery)) {
+        Assertions.assertEquals(vectorSchemaRoot.getRowCount(), 100);
+        Assertions.assertEquals(vectorSchemaRoot.getFieldVectors().size(), 4);
+        for (FieldVector fieldVector : vectorSchemaRoot.getFieldVectors()) {
+          for (int i = 0; i < 100; i++) {
+            Assertions.assertNotNull(fieldVector.getObject(i));
+          }
+        }
+      }
+    }
+  }
```

Comment:
These tests seems to me as integrated tests but instead of using the real db your mocking everything.

Try create more unit testing. There are more functions that is not being tested such as the ones in the VectorSchemaRootWrapper class